server/storage.py

```diff
@@ -148,7 +160,24 @@ def get_blob_url(self, obj_name, container_name=None, timeout=60):
         secure_url = self.get_url(obj_path)
         return "{0}?{1}".format(secure_url, url_kv)
 
-    def get_object_stream(self, object):
-        """ Data stream of the libcloud object.
+    def _generate_azure_signed_url(self, obj_name, expiry_datetime):
+        """ Generates a signed URL compatible with Azure Blob Storage
         """
-        return self.driver.download_object_as_stream(object)
+        # libcloud encodes the access key, so it needs decoding
+        secret_as_string = base64.b64encode(self.driver.secret).decode()
+
+        # azure expiry times need to be ISO8601 (no milliseconds, and with the timezone character)
+        expires = expiry_datetime.replace(microsecond=0).isoformat() + 'Z'
```

Comment:
It looks like here it is assuming Zulu or UTC time zone when local time is being passed to `expiry_datetime`.